packages/salesforcedx-utils/test/jest/cli/cliCommandExecutor.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as cross_spawn from 'cross-spawn';
+import { CliCommandExecutor, Command, TELEMETRY_HEADER } from '../../../src';
+import { CliCommandExecution } from '../../../src/cli/cliCommandExecution';
+import { GlobalCliEnvironment } from '../../../src/cli/globalCliEnvironment';
+jest.mock('cross-spawn');
+jest.mock('../../../src/cli/cliCommandExecution');
+const crossSpawnMocked = jest.mocked(cross_spawn);
+const CliCommandExecutorMock = jest.mocked(CliCommandExecution);
+
+describe('CliCommandExecutor Unit Tests.', () => {
+  const fakeCommand: Command = {
+    command: 'do a thing',
+    args: ['arg1', 'arg2'],
+    toCommand: jest.fn()
+  };
+  const options = {
+    env: {
+      TEST_ENV: 'weAreTestingForSure'
+    },
+    timeout: 2000
+  };
+
+  const globalKey = 'globalKey';
+  const globalValue = 'totallyTrue';
+
+  beforeEach(() => {
+    // Add a global value to the GCE option for processing during creation.
+    GlobalCliEnvironment.environmentVariables.set(globalKey, globalValue);
```

Comment:
this is an exported Map so I decided to just set a value verse mocking the Map functionality.